File: 3227-find-missing-and-repeated-values/find-missing-and-repeated-values.cpp
class Solution {
public:
    vector<int> findMissingAndRepeatedValues(vector<vector<int>>& grid) {


        int n = grid.size();

        vector<int> freq(n * n + 1, 0);
        vector<int> ans;

        for(auto &row: grid) {

            for(auto &col: row) {
                
                if(freq[col]) ans.push_back(col);
                freq[col]++;
            }
        }

        for(int i = 1; i < freq.size(); i++) {

            if(freq[i] == 0) {
                ans.push_back(i);
                break;
            }
        }

        return ans;
        
    }
};

File: 3227-find-missing-and-repeated-values/find-missing-and-repeated-values.js
/**
 * @param {number[][]} grid
 * @return {number[]}
 */
var findMissingAndRepeatedValues = function(grid) {


    const freq = new Array(grid.length * grid.length + 1).fill(0);
    const ans = [];


    for(const row of grid) {

        for(const col of row) {
            freq[col]++;
            if(freq[col] == 2) ans[0] = col;
        }
    }


    for(let i = 1; i < freq.length; i++) {

        if(freq[i] == 0) {
            ans[1] = i;
            break;
        }
    }


    return ans;

    
};